File: internal/content/testdata/bench/csharp/max-partition-sum.cs
public class Solution {
    public long max_partition_sum(long[] arr) {
        long total = 0;
        foreach (var x in arr) total += x;
        long current = 0;
        long ans = 0;
        for (int i = 0; i < arr.Length - 1; i++) {
            current += arr[i];
            long candidate = current > total - current ? current : total - current;
            if (candidate > ans) ans = candidate;
        }
        return ans;
    }
}

File: internal/content/testdata/bench/rust/max-partition-sum.rs
fn max_partition_sum(arr: Vec<i64>) -> i64 {
    let total: i64 = arr.iter().sum();
    let mut current: i64 = 0;
    let mut ans: i64 = 0;
    for i in 0..arr.len().saturating_sub(1) {
        current += arr[i];
        ans = ans.max(current.max(total - current));
    }
    ans
}
